src/main/java/seedu/address/model/medicine/Medicine.java

```diff
@@ -0,0 +1,115 @@
+package seedu.address.model.medicine;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+
+/**
+ * Represents a Medicine in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Medicine {
+
+    public static final String MESSAGE_MEDICINE_NAME_CONSTRAINTS =
+            "Medicine Names should not be empty"; // temporarily this will change soon
+
+    public static final String MESSAGE_SHORT_FORM_CONSTRAINTS =
+            "Medicine short forms should not be empty"; // temporarily this will change soon
+
+
+
+    public final String medicineName;
+
+    public final String shortForm;
+    /**
+     * Every field must be present and not null.
+     */
+    public Medicine(String medicineName) {
+        requireNonNull(medicineName);
+        checkArgument(isValidMedicineName(medicineName), MESSAGE_MEDICINE_NAME_CONSTRAINTS);
+        this.medicineName = medicineName;
+        this.shortForm = "";
+    }
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Medicine(String medicineName, String shorfForm) {
+        requireAllNonNull(medicineName, shorfForm);
+        checkArgument(isValidMedicineName(medicineName), MESSAGE_MEDICINE_NAME_CONSTRAINTS);
+        this.medicineName = medicineName;
+        this.shortForm = shorfForm;
+    }
+
+    public String getMedicineName() {
+        return this.medicineName;
+    }
+
+    public String getShorfForm() {
+        return this.shortForm;
+    }
+
+
+    /**
+     * Returns true if both medicines have the same medicine name.
+     * This defines a weaker notion of equality between two medicines.
+     */
+    public boolean isSameMedicine(Medicine m) {
```

Comment:
This method can be simplified.

Use a guard clause to check if null:
After checking m == this, do
if (m == null) {
  return false;
}

Shortform equivalence is debatably unnecessary - but you can choose to leave it in, perhaps as 'defensive coding'

Perhaps there's some room to combine some of the if clauses